lms/static/js/student_account/views/RegisterView.js

```diff
@@ -95,7 +95,7 @@
 
                     if (this.autoSubmit) {
                         $(this.el).hide();
-                        $('#register-honor_code').prop('checked', true);
+                        $('#register-honor_code, #register-terms_of_service').prop('checked', true);
```

Comment:
This would need to be configurable. We wouldn't want to do this on edx.org.

Reply:
This is already configurable.  This code only applies to identity providers with [`skip_registration_form`|https://github.com/edx/edx-platform/blob/622ee7a7e35e33b198561e525bafd93e61aa82c8/common/djangoapps/third_party_auth/models.py#L134-L141] enabled.  You probably don't have such providers on edx.org.

In the standard configuration of Open edX, the registration form has a single check box to accept the honour code and the terms of service.  This single check box has the id `register-honor_code`, so when `skip_registration_form` is used together with the standard configuration of Open edX, the registration form is indeed completely skipped.  The registration form is configurable, however, and can be configured to also (or only) include a check box to accept the terms of service, which has the id `register-terms_of_service`.  In that configuration, the `skip_registration_form` setting stops working as intended, since the registration form is not actually skipped.